### PR TITLE
Implement CS checking based on the `WP_CLI_CS` ruleset

### LINKED ISSUE #28: Adopt and enforce new `WP_CLI_CS` standard
We have a new PHPCS standard for WP-CLI called `WPCliCS` (props @jrfnl). It is part of the `wp-cli/wp-cli-tests` package starting with version v2.1.0.

To adopt & enforce this new standard, the following actions need to be taken for this repository:

- [x] Create a PR that adds a custom ruleset `phpcs.xml.dist` to the repository

    - Add `phpcs.xml.dist` file
    - Adapt `.distignore` to ignore `phpcs.xml.dist` & `phpunit.xml.dist`
    - Adapt `.gitignore` to ignore `phpunit.xml`, `phpcs.xml` & `.phpcs.xml`
    - Require version `^2.1` of the `wp-cli/wp-cli-tests` as a dev dependency
- [ ] Make any required changes to the code that fail the checks from the above ruleset in separate PRs
- [ ] Merge thre ruleset once all required changes have been processed and merged

A sample PR for a simple repository can be seen here: https://github.com/wp-cli/maintenance-mode-command/pull/3/files

Related https://github.com/wp-cli/wp-cli/issues/5179

### PR DESCRIPTION
**Pull request by thrijith**
Add a PHPCS ruleset using the new WP_CLI_CS standard.

Fixes #28 

Related wp-cli/wp-cli#5179

### CODE REVIEW

File: src/Widget_Command.php [hunk 1]
```diff
@@ -646,6 +643,7 @@ private function sanitize_widget_options( $id_base, $dirty_options, $old_options
 
 		// No easy way to determine expected array keys for $dirty_options
 		// because Widget API dependent on the form fields
+		// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged -- Whitelisting due to above reason.
```
Comment (thrijith) [inline src/Widget_Command.php:652]: Whitelisting here as I was not able to generate an error, keeping the silencing operator due to that.

**Review by schlessera — APPROVED** [review]

File: phpcs.xml.dist [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			<property name="prefixes" type="array">
+				<element value="WP_CLI\Widget"/><!-- Namespaces. -->
+				<element value="wpcli_widget"/><!-- Global variables and such. -->
+			</property>
+		</properties>
+	</rule>
+
+	<!-- Exclude existing classes from the prefix rule as it would break BC to prefix them now. -->
+    <rule ref="WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedClassFound">
+        <exclude-pattern>*/src/(Widget|Sidebar)?_Command\.php$</exclude-pattern>
```
Comment (jrfnl) [inline phpcs.xml.dist]: The question mark should be removed. There is no file called `_Command.php`.

```suggestion
        <exclude-pattern>*/src/(Widget|Sidebar)_Command\.php$</exclude-pattern>
```

File: src/Widget_Command.php [hunk 3]
```diff
@@ -144,22 +140,22 @@ public function add( $args, $assoc_args ) {
 		 * 2. Adding the widget to the sidebar
 		 * 3. Positioning appropriately
 		 */
-		$widget_options = $option_keys = $this->get_widget_options( $name );
+		$widget_options = $this->get_widget_options( $name );
+		$option_keys    = $this->get_widget_options( $name );
```
Comment (jrfnl) [inline src/Widget_Command.php]: No need for the duplicate function call.
```php
		$widget_options = $this->get_widget_options( $name );
		$option_keys    = $widget_options;
```

File: src/Sidebar_Command.php [hunk 4]
```diff
@@ -15,11 +15,7 @@
  */
 class Sidebar_Command extends WP_CLI_Command {
 
-	private $fields = array(
-		'name',
-		'id',
-		'description'
-	);
+	private $fields = [ 'name', 'id', 'description' ];
```
Comment (jrfnl) [inline src/Sidebar_Command.php]: Just wondering why this was changed, previous version or a hybrid would be considered more easily readable.
```suggestion
	private $fields = [
		'name',
		'id',
		'description',
	];
```

File: src/Widget_Command.php [hunk 5]
```diff
@@ -32,12 +32,7 @@
  */
 class Widget_Command extends WP_CLI_Command {
 
-	private $fields = array(
-		'name',
-		'id',
-		'position',
-		'options',
-		);
+	private $fields = [ 'name', 'id', 'position', 'options' ];
```
Comment (jrfnl) [inline src/Widget_Command.php]: Same remark as with the other property.

**Review by jrfnl — COMMENTED** [review]
Just some minor remarks.

File: src/Sidebar_Command.php (hunk 4 above)
Comment (thrijith) [reply src/Sidebar_Command.php]: Since there were not many values, I shifted it to single line, reverted back.

**Review by jrfnl — APPROVED** [review]

**Comment by schlessera** [comment]
Test failures seem unrelated and tracked in #31 & #32. Merging this one anyway.

**Review by schlessera — APPROVED** [review]

**Pull request merged**